go/libraries/doltcore/sqle/writer/prolly_index_writer.go

```diff
@@ -413,3 +413,192 @@ func FormatKeyForUniqKeyErr(key val.Tuple, d val.TupleDesc) string {
 	sb.WriteString("]")
 	return sb.String()
 }
+
+type prollySecondaryIndexPrefixWriter struct {
```

Comment:
Seems like this could just be a branch in the regular index writer. I'm not seeing much difference here for the additional 200 lines of maintenance when some refactoring would work